src/lib/comms/sol-lwm2m.c

```diff
@@ -3605,8 +3854,14 @@ dispatch_notifications(struct sol_lwm2m_client *client,
             resource->path[path_idx]))
             continue;
 
-        r = send_notification_pkt(client, obj_ctx, NULL, -1, obj_ctx->obj_res);
-        SOL_EXP_CHECK(!r, false);
+        ctx.obj_ctx = obj_ctx;
+        ctx.resource_id = -1;
+        r = sol_coap_notify_by_callback(client->coap_server, obj_ctx->obj_res,
```

Comment:
Oops, changed to int.
Now I know why the compiler was complaining about my `SOL_INT_CHECK(r, < 0, false);` and I had to use `if`s. Changed that too, thanks.